cfme/common/vm.py

```diff
@@ -63,6 +63,72 @@ def all_types(template=False):
     return all_types
 
 
+def do_set_retirement_date(view, when=None, offset=None, warn=None, multiple_vms=False):
+    fill_date = None
+    fill_offset = None
+
+    # explicit is/not None use here because of empty strings and dicts
+    if when is not None and offset is not None:
+        raise ValueError('set_retirement_date takes when or offset, but not both')
+    if when is not None and not isinstance(when, (datetime, date)):
+        raise ValueError('when argument must be a datetime object')
+
+    # format the date
+    # needs 4 digit year for fill
+    # displayed 2 digit year for flash message
+    changed = False
+    if offset is not None:
+        fill_offset = {k: v
+                       for k, v in offset.items()
+                       if k in ['months', 'weeks', 'days', 'hours']}
+        # timedelta can take weeks, but not months
+        # copy and pop, only used to generate message, not used for form fill
+        offset_copy = fill_offset.copy()
+        if 'months' in offset_copy:
+            new_weeks = offset_copy.get('weeks', 0) + int(offset_copy.pop('months', 0)) * 4
+            offset_copy.update({'weeks': new_weeks})
+
+        msg_date = datetime.utcnow() + timedelta(**offset_copy)
+        msg_date = msg_date.strftime('%m/%d/%y %H:%M UTC')
+        msg = (f"Retirement dates set to {msg_date}" if multiple_vms else
+               f"Retirement date set to {msg_date}")
+    elif when is not None:
+        fill_date = when.strftime('%m/%d/%Y %H:%M')
+        msg_date = when.strftime('%m/%d/%y %H:%M UTC')
+        msg = (f"Retirement dates set to {msg_date}" if multiple_vms else
+               f"Retirement date set to {msg_date}")
+    else:
+        # clearing retirement date with space in textinput,
+        # using space here as with empty string calendar input is not cleared correctly
+        fill_date = ' '
```

Comment:
Does this trick work on `wharf`? @jawatts is hitting a similar issue with a `TextInput` field.

Reply:
I've been doing my local testing with wharf, and it works for me.